packages/grid/x-data-grid/src/hooks/features/rows/useGridRows.ts

```diff
@@ -105,26 +100,27 @@ const getRowsStateFromCache = (
 export const rowsStateInitializer: GridStateInitializer<
   Pick<DataGridProcessedProps, 'rows' | 'rowCount' | 'getRowId' | 'loading'>
 > = (state, props, apiRef) => {
-  const rowsCache = {
-    state: convertGridRowsPropToState({
-      rows: props.rows,
-      getRowId: props.getRowId,
-      prevState: {
-        value: {
-          idRowsLookup: {},
-          ids: [],
-        },
-        rowsBeforePartialUpdates: [],
+  apiRef.current.rowsCache = convertRowsPropToState({
```

Comment:
If we keep it in the apiRef, what do you think about doing `apiRef.current.caches.rows` ?
Or `apiRef.current.unstable_caches.rows` if we want to keep it private for now.

I would like to move `unstable_sanitizedModelOnLastRowTreeCreation` out of the state (same for the equivalent of the Aggregation)
Maybe it's worth scoping it instead of doing it on the root (same for the DOM refs btw)

For the alternative you gave in the PR description, it don't know how we could adapt it in the future to allow custom plugins with there own cache without touching the `useDataGridXXXComponent` hook.
But as long as the caches system is private, we can rewrite it to be compatible when we know what the plugins will actually look like

Reply:
I went with `unstable_caches`. I hope that theme augmentation works to enrich the cache interface to allow custom plugins to have their own cache.

Reply:
It should work perfectly